lib/opFns.js

```diff
@@ -291,13 +291,28 @@ module.exports = {
   EXTCODEHASH: function (address, runState, cb) {
     var stateManager = runState.stateManager
     address = addressToBuffer(address)
-    stateManager.getContractCode(address, function (err, code) {
+
+    if (runState._precompiled[address.toString('hex')]) {
+      cb(null, new BN(0))
```

Comment:
If non-existent account has a result 0, while empty code has the empty keccak256, why is the hash of the precompile 0?

Reply:
I would think that this is a way to signal that the contract shouldn't be calling `EXTCODEHASH` on a precompile address.

Reply:
I can see how this would be confusing behavior and maybe the hash of a precompile should just be `KECCAK256_NULL`

Reply:
I think it should be the empty code hash for precompiles, since the account does exists.